File: ts/project/.sdk/tm/scala/core/Response.scala
package SCALAPACKAGE.core

import java.util.{Map => JMap}
import java.util.function.Supplier
import SCALAPACKAGE.utility.struct.Struct

// A transport-level response (thin wrapper over the fetcher's map shape).
class Response(resmap: JMap[String, Object]) {

  var status: Int = -1
  var statusText: String = ""
  var headers: Object = null
  var jsonFunc: Supplier[Object] = null
  var body: Object = null
  var err: RuntimeException = null

  locally {
    val s = Struct.getprop(resmap, "status")
    if (s != null) status = Helpers.toInt(s)

    Struct.getprop(resmap, "statusText") match { case st: String => statusText = st; case _ => }

    headers = Struct.getprop(resmap, "headers", null)

    Struct.getprop(resmap, "json") match {
      case jf: Supplier[_] => jsonFunc = jf.asInstanceOf[Supplier[Object]]
      case _ =>
    }

    body = Struct.getprop(resmap, "body", null)

    Struct.getprop(resmap, "err") match {
      case e: RuntimeException => err = e
      case _ =>
    }
  }
}

File: ts/project/.sdk/tm/kotlin/core/Response.kt
package KOTLINPACKAGE.core

import java.util.function.Supplier

import KOTLINPACKAGE.utility.struct.Struct

/** A transport-level response (thin wrapper over the fetcher's map shape). */
@Suppress("UNCHECKED_CAST")
class Response(resmap: Map<String, Any?>?) {

  var status: Int = -1
  var statusText: String = ""
  var headers: Any? = null
  var jsonFunc: Supplier<Any?>? = null
  var body: Any? = null
  var err: RuntimeException? = null

  init {
    val s = Struct.getprop(resmap, "status")
    if (s is Number) {
      this.status = Helpers.toInt(s)
    }

    val st = Struct.getprop(resmap, "statusText")
    if (st is String) {
      this.statusText = st
    }

    this.headers = Struct.getprop(resmap, "headers", null)

    val jf = Struct.getprop(resmap, "json")
    if (jf is Supplier<*>) {
      this.jsonFunc = jf as Supplier<Any?>
    }

    this.body = Struct.getprop(resmap, "body", null)

    val e = Struct.getprop(resmap, "err")
    if (e is RuntimeException) {
      this.err = e
    }
  }
}
